### PR TITLE
add consul config adapter

### PR DESCRIPTION
**Pull request by monchickey**
Add consul implements gcfg.Adapter using consul service.

### CODE REVIEW

File: contrib/config/consul/consul.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+			v = nil
+			return
+		}
+		var ok bool
+		if v, ok = raw.(*api.KVPair); !ok {
+			return
+		}
+
+		if err := c.doUpdate(v.Value); err != nil {
+			g.Log(LoggerName).Errorf(context.Background(),
```
Comment (gqcn) [inline contrib/config/consul/consul.go:149]: Any logging operation please use an interface, and passed implements from user, not using `g.Log` directly.

File: contrib/config/consul/consul.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	"github.com/hashicorp/consul/api/watch"
+)
+
+const LoggerName = "consul-adapter"
+
+// Config is the configuration object for consul client.
+type Config struct {
+	Params api.Config `v:"required"`
+	Path   string     `v:"required"`
+	Watch  bool
```
Comment (gqcn) [inline contrib/config/consul/consul.go:28]: Please add comments for all attributes.

File: contrib/config/consul/consul.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+// Client implements gcfg.Adapter implementing using consul service.
+type Client struct {
+	config Config
+	client *api.Client
+	value  *g.Var
```
Comment (gqcn) [inline contrib/config/consul/consul.go:35]: Please add comments for all attributes.

File: contrib/config/consul/consul.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	"github.com/gogf/gf/v2/os/gcfg"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/api/watch"
+)
+
+const LoggerName = "consul-adapter"
+
+// Config is the configuration object for consul client.
+type Config struct {
+	Params api.Config `v:"required"`
```
Comment (gqcn) [inline contrib/config/consul/consul.go:26]: Is it better using `ConsulConfig` instead of `Params` for this attribute name?

**Review by gqcn — CHANGES_REQUESTED** [review]

**Comment by gqcn** [comment]
@monchickey The CI failed it was missing CI service here: https://github.com/gogf/gf/blob/master/.github/workflows/gf.yml

File: contrib/config/consul/consul.go (hunk 1 above)
Comment (monchickey) [reply contrib/config/consul/consul.go:149]: OK

File: contrib/config/consul/consul.go (hunk 4 above)
Comment (monchickey) [reply contrib/config/consul/consul.go:26]: Yes

**Review by gqcn — CHANGES_REQUESTED** [review]

**Review by gqcn — APPROVED** [review]

**Pull request merged**